synapse/storage/databases/main/room.py

```diff
@@ -1191,6 +1198,63 @@ def _background_add_rooms_room_version_column_txn(txn: LoggingTransaction):
 
         return batch_size
 
+    async def _background_populate_rooms_creator_column(
+        self, progress: dict, batch_size: int
+    ):
```

Comment:
This is not too dissimilar from `_background_add_rooms_room_version_column` which also sets `creator` 🤔 

Does a background update iterate over every single row? I was expecting it to fetch only the rows missing `creator` and fill them in.